cg/meta/demultiplex/utils.py

```diff
@@ -95,3 +93,39 @@ def parse_flow_cell_directory_data(
 ) -> FlowCellDirectoryData:
     """Return flow cell data from the flow cell directory."""
     return FlowCellDirectoryData(flow_cell_path=flow_cell_directory, bcl_converter=bcl_converter)
+
+
+def parse_manifest_file(manifest_file):
+    """Returns a list with the first entry of each row of the given TSV file."""
+    files: List[List[str]] = read_csv(manifest_file, delimiter="\t")
+    file_paths = [Path(file[0]) for file in files]
+    return file_paths
+
+
+def is_file_relevant(file: Path) -> bool:
+    """Returns whether a file is relevant for demultiplexing."""
+    relevant_directories = [DemultiplexingDirsAndFiles.INTER_OP, DemultiplexingDirsAndFiles.DATA]
+    for relevant_directory in relevant_directories:
+        if relevant_directory in file:
+            return True
+    return False
+
+
+def is_syncing_complete(source_directory: Path, target_directory: Path) -> bool:
+    """Returns wheter all relevant files for demultiplexing have been synced from the source to
+    the destination."""
+    manifest_file = Path(source_directory, DemultiplexingDirsAndFiles.FILE_MANIFEST)
+    if not manifest_file.exists():
+        LOG.debug(
+            f"{source_directory} does not contain a "
+            f"{DemultiplexingDirsAndFiles.FILE_MANIFEST} file. Skipping."
+        )
+        return False
+    files_at_source: List[Path] = parse_manifest_file(manifest_file)
+    for file in files_at_source:
+        if is_file_relevant(file) and not Path(target_directory, file).exists():
+            LOG.info(
+                f"All files have not been transferred from {target_directory} to {source_directory}"
+            )
+            return False
```

Comment:
Should we also be dead certain and include size check? https://blog.enterprisedna.co/how-to-get-file-size-in-python-a-quick-guide/

Reply:
Me and @karlnyr did some testing and rsync/lsync only makes the file available once it is fully transferred. And even though a size check on a single file is nice and clean, checking the size of a directory means traversing every subdirectory and summing the byte sizes, which takes a while and is not as clean as just calling one function on one  path. I think this will do the trick, and if not we can add a size check

Reply:
Sounds perfect 👍